api/src/main/java/com/redhat/ipaas/api/v1/rest/KubernetesClientProducer.java

```diff
@@ -15,17 +15,17 @@
  */
 package com.redhat.ipaas.api.v1.rest;
 
-import javax.enterprise.context.ApplicationScoped;
-import javax.enterprise.inject.Produces;
-
 import io.fabric8.kubernetes.client.DefaultKubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 
-@ApplicationScoped
+@Configuration
 public class KubernetesClientProducer {
```

Comment:
Maybe we should also rename from KubernetesClientProducer to KubernetesClientConfiguration.

or 

I would probably ditch that and add `spring-cloud-kubernetes` into the mix and get a fully configurable client for free.